src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -237,6 +237,14 @@ class TreeChecker extends Phase with SymTransformer {
       tree
     }
 
+    /** Check that all methods have either MethodicType or PolyType(MethodicType) */
+    def isMethodType(pt: Type)(implicit ctx: Context): Boolean = pt match {
```

Comment:
Thanks @DarkDimius , `AnnotatedType` is indeed missing here. But there's something more related to default parameters.

Following generated function for default parameters

```
    def copy$default$1[a]: a @uncheckedVariance =
      Some.this.x: a @uncheckedVariance
```

gets following type:

```
PolyType(List(a), List(TypeBounds(TypeRef(ThisType(TypeRef(NoPrefix,scala)),Nothing), TypeRef(ThisType(TypeRef(NoPrefix,scala)),Any))), AnnotatedType(TypeRef(ThisType(TypeRef(ThisType(TypeRef(NoPrefix,<empty>)),Some)),Some$$a),ConcreteAnnotation(Apply(Select(New(TypeTree[TypeRef(ThisType(TypeRef(NoPrefix,unchecked)),uncheckedVariance)]),<init>),List()))))
```

Is it valid for methods of default parameters to take a non-method type?